### PR TITLE
Public url for microservices in the platform but not in studio

### PR DESCRIPTION
**Pull request by freshteapot**
## Summary

- Rename HttpResponseApplications2 to HttpResponseApplication
- Expose the Ingress host + path per customer Tenant ID
- Expose Unique Ingress Paths (microservice paths)

### CODE REVIEW

**Comment by freshteapot** [comment]
# Info
- Linked to https://github.com/dolittle/Studio/pull/139

File: pkg/platform/k8s_repo.go [hunk 1]
```diff
@@ -653,3 +670,94 @@ func (r *K8sRepo) createAnnotationsAndLabels(customerID, customerName, applicati
...
+		for _, rule := range ingress.Spec.Rules {
+			for _, rulePath := range rule.HTTP.Paths {
+				// For now, there is no need to expose it
+				rulePath.Backend = networkingv1.IngressBackend{}
+
+				diffA, _ := json.Marshal(rulePath)
+				exists := funk.Contains(items, func(item networkingv1.HTTPIngressPath) bool {
+					diffB, _ := json.Marshal(item)
+					return string(diffA) == string(diffB)
+				})
```
Comment (joelhoisko) [inline pkg/platform/k8s_repo.go:743]: This led me down the rabbit hole on how to compare structs in golang and I found that structs are comparable with the basic [`==` equality operator](https://www.geeksforgeeks.org/structure-equality-in-golang/). So is there a reason you're going the way of JSONifying the structs first? Like relying on some of the structs JSON annotations to clean up the strings?

**Review by joelhoisko — CHANGES_REQUESTED** [review]
Nice work and tests too :) One question about comparing structs though.

File: pkg/platform/k8s_repo.go (hunk 1 above)
Comment (freshteapot) [reply pkg/platform/k8s_repo.go]: I can't explain it, with proof. So we shall use the "==", its cleaner :).
If I ever stumble upon the reason why, I will come back to you. But for now, I am keen to embrace this as its cleaner.

I thought there was something about pointers, but the documentation tells me I am wrong and running the code in question with some printing confirmed, whatever I am trying to sell, does not apply here :P.

```go
for _, rule := range ingress.Spec.Rules {
	for _, rulePath := range rule.HTTP.Paths {
		// For now, there is no need to expose it
		rulePath.Backend = networkingv1.IngressBackend{}

		diffA, _ := json.Marshal(rulePath)
		fmt.Println("HELLO", len(items))
		items = append(items, rulePath)
		exists := funk.Contains(items, func(item networkingv1.HTTPIngressPath) bool {
			diffB, _ := json.Marshal(item)
			fmt.Println("operator match", rulePath == item)
			fmt.Println("string match", string(diffA) == string(diffB))
			return string(diffA) == string(diffB)
		})

		if exists {
			continue
		}
		items = append(items, rulePath)
	}
}
```

**Review by joelhoisko — APPROVED** [review]

**Pull request merged**